comm/src/main/scala/coop/rchain/comm/errors.scala

```diff
@@ -1,35 +1,38 @@
 package coop.rchain.comm
 
+import java.nio.file._
 import coop.rchain.comm.protocol.routing._
 import cats._, cats.data._, cats.implicits._
 import coop.rchain.catscontrib._, Catscontrib._, ski._
 
 // TODO we need lower level errors and general error, for now all in one place
 // TODO cleanup unused errors (UDP trash)
 sealed trait CommError
```

Comment:
`sealed abstract class` is said to be better

Reply:
how come?

Reply:
It's needed if you want proper stacktraces. Try printing a stacktrace for any of the cause-containing errors and you'll see they're broken and are missing debug info.

We've been through that in rholang.

See InterpreterError and subclasses

```scala
  sealed abstract class InterpreterError(message: String) extends Throwable(message) {

    def this(message: String, cause: Throwable) {
      this(message)
      initCause(cause)
    }
  }
```

Reply:
OTOH you're not extending exception and all stacktraces are printed (if at all) from the underlying exceptions, so I'd stay with a trait here. Unless there's another reason, @dzajkowski?

Reply:
I don't want to block the pr, will link to resource.